base/essentials.jl

```diff
@@ -95,6 +95,7 @@ tail(x::Tuple) = argtail(x...)
 
 convert{T<:Tuple{Any,Vararg{Any}}}(::Type{T}, x::Tuple{Any, Vararg{Any}}) =
     tuple(convert(tuple_type_head(T),x[1]), convert(tuple_type_tail(T), tail(x))...)
+convert{T<:Tuple{Any,Vararg{Any}}}(::Type{T}, x::T) = x
```

Comment:
That seems to be the case. In fact, 0-element tuple conversion appears to have its own method [here](https://github.com/jrevels/julia/blob/master/base/essentials.jl#L61).